cmd/kola/etcdtests/etcdDiscovery.go

```diff
@@ -22,29 +24,40 @@ func etcdDiscovery(cluster platform.Cluster) error {
 	for i, m := range cluster.Machines() {
 		// start etcd instance
 		etcdStart := "sudo systemctl start etcd.service"
-		_, err := m.SSH(etcdStart)
+		//etcdStart := "etcd version"
+		b, err := m.SSH(etcdStart)
 		if err != nil {
 			return fmt.Errorf("SSH cmd to %v failed: %s", m.IP(), err)
 		}
-		fmt.Fprintf(os.Stderr, "etcd instance%d started\n", i)
+		fmt.Fprintf(os.Stderr, "etcd instance%d started%s\n", i, b)
 	}
 
 	fmt.Fprintf(os.Stderr, "Waiting for discovery to finish...\n")
-	time.Sleep(1 * time.Second)
 
-	// get status of etcd instances
-	cmd := cluster.NewCommand("curl", "-L", "http://10.0.0.2:4001/v2/machines")
-	b, err := cmd.Output()
-	if err != nil {
-		return fmt.Errorf("Error requesting cluster members: %v", err)
+	// get status of etcd instances -- retry until satisfied
+	var members []string
+	var success bool
+	for i := 0; i < curlRetries; i++ {
+		time.Sleep(time.Second)
+		cmd := cluster.NewCommand("curl", "http://10.0.0.2:4001/v2/machines")
+		b, err := cmd.Output()
+		if err != nil {
+			continue
+		}
+		members = strings.Split(string(b), ",")
+		// etcd 2.0 listens on 2 ports for clients and returne 2*csize
+		if len(members) == csize || len(members) == csize*2 {
```

Comment:
Curling `/v2/machines` seemed to be the only version agnostic way to easily see that the cluster had formed correctly, but is slightly ugly in that 2.0.x normally returns a list that is double the cluster size.  curling `/v2/members` is 2.0.x only.

I'll check that `etcdctl cluster-health` returns 3 healthy members instead.